src/backend/distributed/commands/trigger.c

```diff
@@ -611,6 +648,64 @@ ErrorOutForTriggerIfNotSupported(Oid relationId)
 }
 
 
+#if (PG_VERSION_NUM < PG_VERSION_15)
+
+/*
+ * ErrorOutIfCloneTrigger is a helper function to error
+ * out if we are trying to rename a child trigger on a
+ * partition of a distributed table.
+ * A lot of this code is borrowed from PG15 because
+ * renaming clone triggers isn't allowed in PG15 anymore.
```

Comment:
if it's not supported, couldn't we just let it error out in ProcessUtility or do we run into trouble before then?

Reply:
> couldn't we just let it error out in ProcessUtility?

For PG15 yes. For earlier versions, no as the error message is confusing and misleading.

Renaming child triggers is prohibited on PG15 only. The change is not backported to earlier versions of PG.

This helper function is only used in earlier versions of PG, where renaming of child triggers are still allowed. The rename works on coordinator shell table, but gives misleading error messages as the triggers on the shards are not found with the expected names.

Reply:
got it, thanks